src/enigma2/Channels.cpp

```diff
@@ -199,5 +202,48 @@ bool Channels::LoadChannels(const std::string groupServiceReference, const std::
   }
 
   Logger::Log(LEVEL_INFO, "%s Loaded %d Channels", __FUNCTION__, GetNumChannels());
+
+  if (Admin::CanUseJsonApi())
+  {
+    //We can use the JSON API so let's supplement the data with provider information
+
+    const std::string jsonURL = StringUtils::Format("%sapi/getservices?provider=1&picon=1&sRef=%s", Settings::GetInstance().GetConnectionURL().c_str(), WebUtils::URLEncodeInline(groupServiceReference).c_str());
+    const std::string strJson = WebUtils::GetHttpXML(jsonURL);  
+    
+    try
+    {
+      auto jsonDoc = json::parse(strJson);
+
+      if (!jsonDoc["services"].empty())
+      {
+        for (json::iterator it = jsonDoc["services"].begin(); it != jsonDoc["services"].end(); ++it) 
```

Comment:
`for (const auto& jsonChannel : jsonDoc["services"])`?

Reply:
With the .items() it works.